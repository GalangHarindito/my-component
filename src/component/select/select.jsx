import React from "react";
import PropTypes from "prop-types";
import "./select.scss";

export const SelectInput = ({
  label,
  name,
  defaultValue,
  options,
  size,
  style,
  onChange,
  error,
  required,
  ...props
}) => {
  return (
    <>
      <div className="wrapper-select">
        {label && (
          <label className="label-input" htmlFor="input">
            {label}
            {required && <span className="asterix">*</span>}
          </label>
        )}
        <select
          className={[
            "storybook-select",
            `storybook-size-${size}`,
            `${error && `storybook-error-boundary`}`,
          ].join(" ")}
          name={name}
          onChange={onChange}
          {...props}
        >
          <option className="storybook-option-default" value="">
            {defaultValue}
          </option>
          {options &&
            options.map((option) => {
              return (
                <option className="storybook-option" value={option.value}>
                  {option.icon && option.icon} <span>{option.name}</span>
                </option>
              );
            })}
        </select>
        {error && (
          <div className="wrapper-error-message">
            <p className="error-input">{error}</p>
          </div>
        )}
      </div>
    </>
  );
};

SelectInput.prototype = {
  label: PropTypes.string,
  defaultValue: PropTypes.string,
  onChange: PropTypes.func,
  size: PropTypes.string,
  error: PropTypes.string,
  required: PropTypes.bool,
  options: PropTypes.arrayOf(PropTypes.object),
};

SelectInput.defaultProps = {
  onChange: undefined,
  label: "Select Input",
  defaultValue: "Select Input",
  size: "large",
  error: "",
  required: false,
  options: [
    { icon: "ðŸ¤£", value: "test1", name: "test1" },
    { icon: "ðŸ˜†", value: "test2", name: "test2" },
    { icon: "ðŸ¥°", value: "test3", name: "test3" },
  ],
};
