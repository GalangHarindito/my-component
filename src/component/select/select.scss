@import "../../utils/color.scss";

.wrapper-select {
  display: flex;
  flex-direction: column;

  .storybook-select {
    font-family: "Poppins", sans-serif;
    height: 2rem;
    font-size: 1rem;
    outline: none;
    border: none;
    border-bottom: 1.8px solid #515151;
    background: transparent;
    width: 100%;
    font-size-adjust: 0.3;
    &:focus {
      border-bottom: 2.2px solid #515151;
    }
    &::placeholder {
      font-size: 0.2rem;
    }

    option{
      font-family: "Poppins", sans-serif;
      font-size: 1rem;
      padding: 0 10rem;
      cursor: pointer;
    }
  }
  .storybook-error-boundary {
    border-bottom: 2.2px solid $Error-red-color-text;
  }

  .storybook-size-large {
    width: 100%;
  }

  .storybook-size-medium {
    width: 50%;
  }

  .label-input {
    margin: 0.5rem 0;
    font-size: 1rem;
    font-weight: 500;
  
    .asterix{
      color: $Error-red-color-text;
      font-size: 0.8rem;
      margin-left: 0.2rem;
    }
  }

  .wrapper-error-message {
    .error-input {
      margin: 0.2rem 0;
      color: $Error-red-color-text;
      font-weight: 600;
      font-size: 0.9rem;
      padding-left: 0.2rem;
    }
  }
}
