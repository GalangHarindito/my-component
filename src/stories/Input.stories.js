import { Input } from "../component/input/Input";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
export default {
  title: 'Component/Input',
  component: Input,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes

  argTypes:{
    onChange: {
      description: 'Handle behaviour when input event',
      action: 'type',
    },
    placeholder: {
      description: 'Name for placeholder',
      action: 'string',
    },
    label: {
      description: 'Name for input label',
      action: 'string',
    },
    size: {
      description: 'Size for input',
      action: 'string',
    },
    error: {
      description: 'Error message if any',
      action: 'string',
    }
  }
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Large = {
  args: {
    size: 'large',
    label: 'Label Text',
    placeholder: 'Input Name',
    error: ''
  },
};

export const Medium = {
    args: {
      size: 'medium',
      label: 'Label Text',
      placeholder: 'Input Name',
      error: ''
    },
  };