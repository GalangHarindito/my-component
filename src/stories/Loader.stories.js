import { Loader } from "../component/spinner/loader";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
export default {
  title: 'Component/Loader',
  component: Loader,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes

  argTypes:{
    size: {
      description: 'Size for loader',
      action: 'string',
    }
  }
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Large = {
  args: {
    size: 'large',
  },
};

export const Medium = {
    args: {
      size: 'medium',
    },
  };

  export const Small = {
    args: {
      size: 'small',
    },
  };