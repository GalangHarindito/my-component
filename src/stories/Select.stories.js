import { SelectInput } from "../component/select/select";

export default {
    title: 'Component/Select',
    component: SelectInput,
    parameters: {
      // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
      layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
    tags: ['autodocs'],
    // More on argTypes: https://storybook.js.org/docs/api/argtypes
  
    argTypes:{
        onChange: {
            description: 'Handle behaviour when input event',
            action: 'type',
          },
          label: {
            description: 'Name for input label',
            action: 'string',
          },
          size: {
            description: 'Size for input',
            action: 'string',
          },
          error: {
            description: 'Error message if any',
            action: 'string',
          },
          required: {
            description: 'Required field',
            action: 'boolean',
          },
    }
  };
  
  // More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
  
  export const Select = {
    args: {
      size: 'large',
      label: 'Label Text',
      defaultValue: ' Input Value',
      error: '', 
      required: false,
      options: [
        { icon: "ðŸ¤£", value: "test1", name: "test1" },
        { icon: "ðŸ˜†", value: "test2", name: "test2" },
        { icon: "ðŸ¥°", value: "test3", name: "test3" },
      ]
    },
  };