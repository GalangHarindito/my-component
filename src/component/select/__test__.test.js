import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../../stories/Select.stories';
import { SelectInput } from './select'; // import all stories from the stories file

const { Select } = composeStories(stories);

test('Make sure input UI does not change unexpectedly', () => {
    const { container } = render(<Select />)
    expect(container).toMatchSnapshot()
}) 