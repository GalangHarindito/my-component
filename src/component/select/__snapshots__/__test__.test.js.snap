// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure input UI does not change unexpectedly 1`] = `
<div>
  <div
    class="wrapper-select"
  >
    <label
      class="label-input"
      for="input"
    >
      Label Text
    </label>
    <select
      class="storybook-select storybook-size-large "
    >
      <option
        class="storybook-option-default"
        value=""
      >
         Input Value
      </option>
      <option
        class="storybook-option"
        value="test1"
      >
        ðŸ¤£
         
        <span>
          test1
        </span>
      </option>
      <option
        class="storybook-option"
        value="test2"
      >
        ðŸ˜†
         
        <span>
          test2
        </span>
      </option>
      <option
        class="storybook-option"
        value="test3"
      >
        ðŸ¥°
         
        <span>
          test3
        </span>
      </option>
    </select>
  </div>
</div>
`;
